src/js/controller.js

```diff
@@ -28,38 +28,30 @@ export default class Controller extends Component {
         this.confirmationModal = new ConfirmationModal(this, ConfirmationModal.COOKIE_NAMES.DATA_MAPPER_VERSION_SELECTION);
 
         $(window).on('hashchange', () => {
-            // On every hash change the render function is called with the new hash.
-            // This is how the navigation of our app happens.
-            this.api.cancelAndInitAbortController();
             const hash = decodeURI(window.location.hash);
             let parts = hash.split(':')
-            let payload = null;
-
+            this.api.cancelAndInitAbortController();
             if (parts[0] == '#geo') {
-                payload = self.changeGeography(parts[1])
-                self._shouldMapZoom = true;
+                this.changeHash(parts[1], true);
+            }
 
-            } else if (parts[0] == '#logout') {
+            if (parts[0] == '#logout') {
                 self.onLogout();
-            } else {
-                //if a category nav is clicked, the hash becomes something like #divId, in that case it should behave same as if hash == ''
-                const areaCode = this.config.rootGeography;
-                self._shouldMapZoom = false;
-                payload = self.changeGeography(areaCode)
             }
-
-            self.onHashChange(payload);
         });
 
         window.addEventListener('popstate', (event) => {
             let hiddenIndicators = [], filteredIndicators = [];
+            let areaCode = '';
```

Comment:
null is often a better default value to distinguish "hasn't been set" from "bad input". No need to change here - just noting in case it's useful in future.